src/accountsdb/snapshots.zig

```diff
@@ -52,16 +52,64 @@ pub const Stakes = struct {
 };
 
 pub const VoteAccounts = struct {
-    vote_accounts: HashMap(Pubkey, struct { u64, Account }),
+    vote_accounts: HashMap(Pubkey, struct { u64, VoteAccount }),
 
     staked_nodes: ?HashMap(
         Pubkey, // VoteAccount.vote_state.node_pubkey.
         u64, // Total stake across all vote-accounts.
     ) = null,
 
     pub const @"!bincode-config:staked_nodes" = bincode.FieldConfig(?HashMap(Pubkey, u64)){ .skip = true };
+
+    pub fn stakedNodes(self: *@This(), allocator: std.mem.Allocator) !*const HashMap(Pubkey, u64) {
```

Comment:
You're right that I made a mistake here.

The goal is for this to be a getter. It is intended to extract the data from the internal state and return it. You should be able to call it many times successfully, and get the staked nodes every time. Since the extraction process is nontrivial, it is memoized. That's the whole idea of the staked_nodes field. It's basically just a cache for this function.

The mistake I made is that in the event of `self.staked_nodes != null`, it should return a pointer to the existing hashmap.

Reply:
76a33a613df464cbb41abc3844ce07cc1cf538a4